text/0001-script-based-tokenizer.md

```diff
@@ -0,0 +1,318 @@
+- Title: Script Based Tokenizer
+- Start Date: 2020-10-27
+- specification PR: meilisearch/specifications#2
+- Meilisearch Issue: meilisearch/Meilsearch#624
+
+## Feature Description and Interaction
+
+### Summary
+
+The first step of document indexing in the Meilisearch engine is tokenization. Tokenization in the action of taking a sentence and splitting in units of language called tokens. The tokenization task is highly language dependant and is a critical factor in the quality of the search results.
+
+### Motivation
+
+We want to provide our users with an always improved searching experience. For that matter, it is critical for us to improve the performance of our tokenizer, and to provide better support for multilingual tokenization.
+
+### Prior Art and R&D
+
+**tokenization:**
+- > **[unicode-segmentation](https://github.com/unicode-rs/unicode-segmentation):**
+  > tokenizer which follow the [Standard Annex #29: Unicode Text Segmentation](http://www.unicode.org/reports/tr29/),
+  > this tokenizer seems promising for Latin scripts.
+- > **[Jieba](https://github.com/messense/jieba-rs):**
+  > tokenizer specialized in Chinese languages
+- > **[Lindera](https://github.com/lindera-morphology/lindera):**
+  > Japanese and Korean
+
+**lang/script detection:**
+- > **[whatlang](https://github.com/greyblake/whatlang-rs):**
+  > whatlang is able to detect script or/and language in a text,
+  > language detection is low but script is acceptable.
+  > note: Sonic also uses whatlang to peform the tokenization, it could be interesting to checkout how they do it.
+- > **toku (@qdequele):**
+  > in a R&D project, @qdequele was able to detect language based on stop word distribution in a text.
+  > If, in a latin script, there is lot of French stop words then the text language is probably french.
+
+**other solution that advertise multilingual support:**
+- Sonic uses whatlang to peform the tokenization, it could be interesting to checkout how they do it: https://github.com/valeriansaliou/sonic/tree/master/src/lexer
```

Comment:
```suggestion
- Sonic uses whatlang to perform the tokenization, it could be interesting to checkout how they do it: https://github.com/valeriansaliou/sonic/tree/master/src/lexer
```